internal/services/post_service.go

```diff
@@ -265,3 +266,27 @@ func (service *PostService) GetPostsPersonalFeed(username string, lastPostId str
 
 	return &feed, nil, http.StatusOK
 }
+
+func (service *PostService) DeletePost(postId uuid.UUID, username string) (*customerrors.CustomError, int) {
+	// Find post by ID
+	post, err := service.postRepo.GetPostById(postId.String())
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return customerrors.PostNotFound, http.StatusNotFound
+		}
+		return customerrors.InternalServerError, http.StatusInternalServerError
+	}
+
+	// Check if the requesting user is the author of the post
+	if post.Username != username {
+		return customerrors.UserUnauthorized, http.StatusUnauthorized
+	}
```

Comment:
Hier ist der Fehler, der den Test zum fehlschlagen bringt:

Änderungen: neuen, vorläufigen Custom Error anlegen (Bsp. siehe unten) und http.StatusForbidden zurückgeben
PostDeleteNotAuthorized = &CustomError{
		Message: "You can only delete your own posts: to be defined",
		Code:    "",
	}